pkg/deploy/lattice/target_group_synthesizer.go

```diff
@@ -271,8 +271,10 @@ func (t *targetGroupSynthesizer) SynthesizeSDKTargetGroups(ctx context.Context)
 				glog.V(6).Infof("tgname %v is not used by httproute %v\n", tgName, httpRoute)
 
 			} else {
-
-				isUsed := t.isTargetGroupUsedByaHTTPRoute(ctx, tgName, httpRoute)
+				// We have finished rule reconciliation at this point.
+				// If a target group under HTTPRoute does not have any service, it is stale.
+				isUsed := t.isTargetGroupUsedByaHTTPRoute(ctx, tgName, httpRoute) &&
+					len(sdkTG.getTargetGroupOutput.ServiceArns) > 0
```

Comment:
I'm having a hard time understanding why lines 270-288 matter, looks like we're just using it for logging or is there some side effect happening I can't see?

Maybe don't worry about it, this super long function could really use some refactoring.